src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -576,7 +579,7 @@ private void OnEnvDTEProjectRemoved(EnvDTE.Project envDTEProject)
 
         private void OnEnvDTEProjectAdded(EnvDTE.Project envDTEProject)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
```

Comment:
Consider this case, you create a solution and then add a project from some VS API, Open solution event and Project Add event are all started from background thread, solution open event start first, then GetAllDTEproject, get 0 project(project is not added yet),skip add projects. then Project Add start, since solution open is still in process, _solutionOpenedRaised is false, Add project event skip. then there are 0 project in nuget cache